vars/retrievePreviousBuild.groovy

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+/** Library to call build.sh script (builds OpenSearch and OpenSearch Dashboards distribution)
+ @param Map args = [:] args A map of the following parameters
+ @param args.inputManifest <required> - Relative path to input manifest containing all the components to build.
+ @param args.platform <required> - Platform of previous build to retrieve.
+ @param args.architecture <required> - Architecture of previous build to retrieve.
+ @param args.distribution <required> - Distribution of previous build to retrieve.
+ @param args.incremental <required> - String value contains the build number of previous build or the latest.
+ */
+void call(Map args = [:]) {
+    def lib = library(identifier: 'jenkins@jenkins_incremental', retriever: legacySCM(scm))
+    def inputManifestObj = lib.jenkins.InputManifest.new(readYaml(file: args.inputManifest))
+
+    def DISTRIBUTION_JOB_NAME = "${JOB_NAME}"
```

Comment:
Do you need to ensure it is the correct job name?

Reply:
Or just use the lib call to parse as arg?

Reply:
This will take any job name where this is run? If someone wants runs this lib in another job, this might cause issue. Take it as an optional arg and default to ${JOB_NAME}

Reply:
@gaiksaya Done in the latest commit.